src/QueryM.purs

```diff
@@ -283,6 +290,30 @@ getDatumsByHashes :: Array DataHash -> QueryM (Map DataHash Datum)
 getDatumsByHashes hashes = unwrap <$> do
   mkDatumCacheRequest DcWsp.getDatumsByHashesCall _.getDatumsByHashes hashes
 
+awaitTxConfirmed :: TxHash -> QueryM Unit
+awaitTxConfirmed = awaitTxConfirmedWithTimeout (Seconds infinity)
+
+awaitTxConfirmedWithTimeout :: Seconds -> TxHash -> QueryM Unit
```

Comment:
I don't want to hold this PR up for this, but ideally this would take a number of `Slot`s to wait instead of seconds